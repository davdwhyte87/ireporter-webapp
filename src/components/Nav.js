import React from 'react'
import ReactDOM from 'react-dom'

class Nav extends React.Component{
  authCheckUser() {
    const token = localStorage.getItem('token');
    if (!token) {
       return false;
    } else {
      return true;  
    }
  }
  toggleNav (e) {
    console.log(e.target.parentElement)
    let navbar = e.target.parentElement
    if (navbar.className === 'navbar') {
        navbar.className += ' responsive';
    } else {
        navbar.className = 'navbar';
    }
  }
  render() {
    if(this.authCheckUser()) {
      const user = JSON.parse(localStorage.getItem('user'));
      return(
        <div>
          <a href="#" onClick={toggleNav()} className="nav-mobile">☰ menu</a>
          <ul itemID="l-nav" >
              <li><a class="active" href="index.html">Home</a></li>
              <li><a href="records.html">Records</a></li>
              <li><a href="create-record.html">Create report</a></li>
              <li><a href="profile.html">Profile</a></li>
          </ul>
          <ul id="r-nav">
              <li><a href="#"  onclick="logout()">Logout <i class="fa fa-fw fa-sign-out"></i> </a></li>
              <li><a href="#" class="nav-name">{user.firstname} </a></li>
          </ul>
        </div>
      )
    }
    return (
      <div>
        <nav className="navbar" id="navbar">
          <a href="#" onClick={this.toggleNav} className='nav-mobile'>☰ menu</a>
          <ul id="l-nav">
              <li><a className="active" href="index.html">Home</a></li>
              <li><a href="sign-in.html">Signin</a></li>
              <li><a href="sign-up.html">Signup</a></li>
          </ul>
          <ul id="r-nav">
          </ul>  
        </nav>
      </div>
    )
  }
}

export default Nav