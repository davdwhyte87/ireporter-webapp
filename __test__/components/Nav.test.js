import React from 'react';
import { mount } from 'enzyme';
import Nav from '../../src/components/Nav';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

describe('Nav test', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    recordsReducer: {
      records: []
    }
  });
  it('should render without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    </Provider>
    );
    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
  it('should match snapshot', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    </Provider>
    );
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  })
});