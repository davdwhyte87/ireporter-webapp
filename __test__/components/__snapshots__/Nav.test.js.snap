// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav test should match snapshot 1`] = `
<div>
  <nav
    className="navbar"
    id="navbar"
  >
    <a
      className="nav-mobile"
      href="#"
      onClick={[Function]}
    >
      â˜° iReporter
    </a>
    <ul
      id="l-nav"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          iReporter
        </a>
      </li>
      <li>
        <a
          href="/signin"
          onClick={[Function]}
        >
          Signin
        </a>
      </li>
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
    </ul>
    <ul
      id="r-nav"
    />
  </nav>
</div>
`;
